extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSource.java

```diff
@@ -107,7 +109,9 @@ private List<S3Object> fetchPrefixedS3Objects() {
 
             var response = client.listObjectsV2(listObjectsRequest);
 
-            s3Objects.addAll(response.contents());
+            Predicate<S3Object> isFile = object -> !object.key().endsWith("/");
```

Comment:
this can be a class field or constant, so it won't be recreated each time

Reply:
addressed in [a6645ac](https://github.com/eclipse-edc/Technology-Aws/pull/417/commits/a6645acb2ef49aa9c677606e86a50dda6b76f3e2)